app/javascript/crayons/MultiInput/MultiInput.jsx

```diff
@@ -0,0 +1,113 @@
+import { h, Fragment } from 'preact';
+import { useRef, useState } from 'preact/hooks';
+
+const KEYS = {
+  ENTER: 'Enter',
+  COMMA: ',',
+  SPACE: ' ',
+};
+// TODO: think about how this may change based on
+// a different usage. We will most likely want this to be passed in as a prop.
+const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9@.])/;
+
+/**
+ * Component allowing users to add multiple inputs that gets displayed as desctructive pills
+ *
+ * @param {Object} props
+ * @param {string} props.placeholder Input placeholder text
+ */
+
+export const MultiInput = ({ placeholder }) => {
+  const inputRef = useRef(null);
+  const [items, setItems] = useState([]);
+
+  const handleBlur = ({ target: { value } }) => {
+    addItemToList(value);
+    clearSelection();
+  };
+
+  const handleKeyDown = (e) => {
+    switch (e.key) {
+      case KEYS.SPACE:
+      case KEYS.ENTER:
```

Comment:
I'm not sure we want the entry to be selected on `Enter` press 🤔  We allow this in the tag multi-select because it follows the combobox pattern where you scroll down with the keyboard arrows to select an option, but for this `MultiInput` we are following default input behaviour (which would be to submit the form on Enter press if it's the last field in the form).

IDK like I say I'm not sure - what do you think?

Reply:
I used Gmail as an inspiration here and looked at the way that adding email addresses when composing a mail on gmail. They seem to select the entry on enter, space, comma and tab. Aside note: I didn't implement the select on tab but would be curious to get your thoughts? 

To me it feels intuitive and quite nice to press enter and have it selected (i.e. move to a pill). However, from an accessibility point of view, if we think it's conflicting with natural behaviour the we should remove it. 

TLDR;I think its nice for the entry to be selected on Enter but if we think its conflicting with natural accessible behaviour then I can remove it.

Reply:
That's good context! I think on balance let's leave it then because I can also see maintaining the default behaviour having potential for confusion (we don't want someone to think it works in the Gmail way, and then accidentally submit the form!)

I don't think we need to do anything on Tab, because Tab key moves the user out of the input and triggers the `onBlur` which selects the current text anyway 🙌